lib/elixir_boilerplate_graphql/operation_name_logger.ex

```diff
@@ -0,0 +1,14 @@
+defmodule ElixirBoilerplateGraphQL.OperationNameLogger do
+  @behaviour Absinthe.Middleware
+
+  def call(resolution, _opts) do
+    current_operation = Enum.find(resolution.path, &current_operation?/1)
+
+    Logger.metadata(graphql_operation_name: current_operation.name)
+
+    resolution
+  end
+
+  defp current_operation?(%Absinthe.Blueprint.Document.Operation{current: true}), do: true
+  defp current_operation?(_), do: false
```

Comment:
> No it can’t 🙂 There’s always an operation in the resolution path, even if it has no explicit name.

I know, this is what I said. My point was about being more explicit on the fact it will always return an `%Operation{}` even if `Enum.find/2` "can" return `nil`.

```suggestion
  def call(resolution, _opts) do
    %Absinthe.Blueprint.Document.Operation{name: name} = Enum.find(resolution.path, &(&1.current))

    Logger.metadata(graphql_operation_name: name)

    resolution
  end
```

Reply:
A `with` statement might even be a better construct to express the logging intention 🤔

Reply:
I like your suggestion but there’s a chance for a crash too if there’s an item in `resolution.path` which has a truthy `current` key.

I think I would keep the `current_operation?/1` function but be more explicit in the matching, even if it means duplicating `%Absinthe.Blueprint.Document.Operation`. What do you think?

```suggestion
  def call(resolution, _opts) do
    %Absinthe.Blueprint.Document.Operation{name: name} = Enum.find(resolution.path, &current_operation?/1)

    Logger.metadata(graphql_operation_name: name)

    resolution
  end

  defp current_operation?(%Absinthe.Blueprint.Document.Operation{current: true}), do: true
  defp current_operation?(_), do: false
```

Reply:
You’re right, I think adding a `with` improves the code a bit since it makes it very clear that `Logger.metadata` will never be called if it doesn’t have a propre operation name 👌 

```elixir
def call(resolution, _opts) do
  with %Absinthe.Blueprint.Document.Operation{name: name} <- Enum.find(resolution.path, &current_operation?/1) do
    Logger.metadata(graphql_operation_name: name)
  end

  resolution
end
```

Reply:
I could actually take it a bit further to explicitely log when no operation name is used:

```elixir
with %Absinthe.Blueprint.Document.Operation{name: name} when not is_nil(name) <- Enum.find(resolution.path, &current_operation?/1) do
  Logger.metadata(graphql_operation_name: name)
else
  _ -> Logger.metadata(graphql_operation_name: "__NONE__")
end
```

@gcauchon @simonprev @EtienneStPierre Think that would be useful?